GamepadEventDrivenInputAPI/explainer.md

```diff
@@ -0,0 +1,237 @@
+
+- `buttonsPressed` and `buttonsReleased`: Indices of buttons whose pressed state transitioned (from pressed to released or vice versa).
+
+- `gamepadSnapshot`: A frozen (read-only) snapshot of the gamepad’s state at the moment the input was received. It includes all axes, buttons, ID, index, and timestamp, and does not update after the event is dispatched.
+
+A new `rawgamepadinputchange` event is dispatched for every gamepad input state change, without delay or coalescing, enabling latency-sensitive applications such as: rhythm games, cloud gaming, or real-time multiplayer scenarios, to respond immediately and accurately to input.
+
+## Example `rawgamepadinputchange` Event
+```js
+rawgamepadinputchange {
+  type: "rawgamepadchange",
+
+  // Snapshot of the gamepad's state at the moment the event was generated.
+  gamepadSnapshot: Gamepad {
+    id: "Xbox Wireless Controller (STANDARD GAMEPAD Vendor: 045e Product: 02fd)",
+    index: 0,
+    connected: true,
+    mapping: "standard",
+    buttons: [
+      // Index 0 - button A pressed.
+      { pressed: true, value: 1.0 },
+      // Index 1 - button B released.
+      { pressed: false, value: 0.0 },
+      // Index 2 - analog button (e.g., triggers).
+      { pressed: false, value: 0.5 },
+      ...
+    ],
+    // [left stick X, left stick Y, right stick X, right stick Y].
+    axes: [0.25, -0.5, 0.0, 0.0],
+    timestamp: 9123456.789
+  },
+
+  // Left stick X and Y moved since last event.
+  axesChanged: [0, 1],
+  // Button index 0 was pressed and button index 1 released, button index 2 value changed.
+  buttonsValueChanged: [0, 1, 2],
+  // Button index 0 pressed.
+  buttonsPressed: [0],
+  // Button index 0 released.
+  buttonsReleased: [1]
+}
+```
+## Proposed IDL
+```
+[Exposed=Window]
+partial interface Gamepad : EventTarget {
+  attribute EventHandler onrawgamepadinputchange;
+};
+
+```
+### `RawGamepadInputChangeEvent` interface IDL, used for `rawgamepadinputchange`.
+```
+// Inherits `target` from Event, which refers to the live Gamepad.
+[Exposed=Window]
+interface RawGamepadInputChangeEvent : Event {
+  constructor(DOMString type, optional RawGamepadInputChangeEventInit eventInitDict = {});
+
+  // Immutable snapshot of gamepad state at time of event dispatch.
+  readonly attribute Gamepad gamepadSnapshot;
+
+  readonly attribute FrozenArray<unsigned long> axesChanged;
+  readonly attribute FrozenArray<unsigned long> buttonsValueChanged;
+  readonly attribute FrozenArray<unsigned long> buttonsPressed;
+  readonly attribute FrozenArray<unsigned long> buttonsReleased;
+};
+
+dictionary RawGamepadInputChangeEventInit : EventInit {
+  required Gamepad gamepadSnapshot;
+  FrozenArray<unsigned long> axesChanged = [];
+  FrozenArray<unsigned long> buttonsValueChanged = [];
+  FrozenArray<unsigned long> buttonsPressed = [];
+  FrozenArray<unsigned long> buttonsReleased = [];
+};
+```
+##  Developer code sample
```

Comment:
We should move this to be directly after the Proposed Approach.  WebIDL is optional for explainers -- like an appendix so let's put that toward the bottom.